collector/diskstats_linux.go

```diff
@@ -227,6 +309,62 @@ func (c *diskstatsCollector) Update(ch chan<- prometheus.Metric) error {
 			}
 			ch <- c.descs[i].mustNewConstMetric(val, dev)
 		}
+
+		if fsType := info[udevIDFSType]; fsType != "" {
+			ch <- c.filesystemInfoDesc.mustNewConstMetric(1.0, dev,
+				fsType,
+				info[udevIDFSUsage],
+				info[udevIDFSUUID],
+				info[udevIDFSVersion],
+			)
+		}
+
+		if name := info[udevDMName]; name != "" {
+			ch <- c.deviceMapperInfoDesc.mustNewConstMetric(1.0, dev,
+				name,
+				info[udevDMUUID],
+				info[udevDMVGName],
+				info[udevDMLVName],
+				info[udevDMLVLayer],
+			)
+		}
+
+		if ata := info[udevIDATA]; ata != "" {
+			for attr, desc := range c.ataDescs {
+				str, ok := info[attr]
+				if !ok {
+					level.Debug(c.logger).Log("msg", "Udev attribute does not exist", "attribute", attr)
+					continue
+				}
+
+				if value, err := strconv.ParseFloat(str, 64); err == nil {
+					ch <- desc.mustNewConstMetric(value, dev)
+				} else {
+					level.Error(c.logger).Log("msg", "Failed to parse ATA value", "err", err)
+				}
+			}
+		}
 	}
 	return nil
 }
+
+func udevDeviceInformation(major, minor uint32) (udevInfo, error) {
+	filename := fmt.Sprintf("/run/udev/data/b%d:%d", major, minor)
+
+	data, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer data.Close()
+
+	info := make(udevInfo)
+
+	scanner := bufio.NewScanner(data)
+	for scanner.Scan() {
+		if name, value, found := strings.Cut(scanner.Text(), "="); found {
```

Comment:
Apparently `strings.Cut()` was added in Go 1.18. Any plans to upgrade soon?

Reply:
We typically support two Go releases. We're building with 1.18, but still support building with Go 1.17.

So, I guess it'll have to wait till Go 1.19. (See `go.mod` for the minimum required Go version).

Reply:
Noted, and fixed in 0da4dda.